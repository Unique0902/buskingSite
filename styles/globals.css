@tailwind base;
@tailwind components;
@tailwind utilities;
/* TODO: 글꼴 불러오는 로딩 줄이거나 없애기 */
* {
  box-sizing: border-box;
  font-family: 'Pretendard';
}

@font-face {
  font-family: 'Pretendard';
  font-weight: 45 920;
  font-style: normal;
  font-display: swap;
  src: url('../assets/fonts/PretendardVariable.woff2')
    format('woff2-variations');
}

li {
  list-style: none;
}
body {
  margin: 0;
  @apply text-black dark:text-white dark:bg-slate-600;
}
p,
h1,
h2 {
  margin: 0;
}
ul {
  padding: 0;
}
