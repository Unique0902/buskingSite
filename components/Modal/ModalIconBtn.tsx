import React, {
  createContext,
  ReactNode,
  useContext,
  useEffect,
  useRef,
  useState,
} from 'react';

import Modal from './Modal';
import { iconName } from '../../assets/icon/constants';
import Icon from '../../assets/icon/icon';
import { useDarkModeContext } from '../../context/DarkModeContext';
import { color } from '../../styles/theme';

type Props = {
  icon: iconName;
  children: ReactNode;
};

type ContextProps = {
  setIsOpenModal: React.Dispatch<React.SetStateAction<boolean>>;
};

const ModalContext = createContext<ContextProps>({} as ContextProps);

const ModalIconBtn = ({ icon, children }: Props) => {
  const [isOpenModal, setIsOpenModal] = useState<boolean>(false);
  const { isDarkMode } = useDarkModeContext();
  const modalRef = useRef<HTMLDivElement>(null);
  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target as Node)
      ) {
        setIsOpenModal(false);
      }
    };
    document.addEventListener('mousedown', handleClickOutside);
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [modalRef, setIsOpenModal]);

  return (
    <ModalContext.Provider value={{ setIsOpenModal }}>
      <button
        onClick={() => {
          setIsOpenModal(true);
        }}
        className='p-2 bg-white border border-black rounded-full dark:bg-slate-800 hover:opacity-70 dark:border-white'
      >
        <Icon
          size={25}
          icon={icon}
          color={isDarkMode ? color.gray_200 : color.gray_900}
        />
      </button>
      {isOpenModal && <Modal modalRef={modalRef}>{children}</Modal>}
    </ModalContext.Provider>
  );
};

export default ModalIconBtn;

export function useModalContext() {
  return useContext(ModalContext);
}
