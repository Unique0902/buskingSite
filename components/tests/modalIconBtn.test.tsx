import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { expect, describe, it } from '@jest/globals';
import ModalIconBtn from '../Modal/ModalIconBtn';

describe('modalIconBtn component test', () => {
  it('modalIconBtn renders correctly', () => {
    const component = renderer.create(
      <ModalIconBtn icon='ArrowDown'>
        <ModalIconBtn.Inner>
          <div>inner</div>
        </ModalIconBtn.Inner>
      </ModalIconBtn>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('click modalIconBtn and show inner box correctly', async () => {
    render(
      <ModalIconBtn icon='ArrowDown'>
        <ModalIconBtn.Inner>
          <div>inner</div>
        </ModalIconBtn.Inner>
      </ModalIconBtn>
    );

    expect(screen.queryByText('inner')).toBeNull();
    const btn = screen.getByRole('button');
    await userEvent.click(btn);
    expect(screen.getByText('inner')).toBeInTheDocument();
  });
});
