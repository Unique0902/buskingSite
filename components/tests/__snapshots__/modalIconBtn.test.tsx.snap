// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modalIconBtn component test modalIconBtn renders correctly 1`] = `
<button
  className="p-2 bg-white border border-black rounded-full dark:bg-slate-800 hover:opacity-70 dark:border-white"
  onClick={[Function]}
>
  <svg
    fill="#191F28"
    height={25}
    viewBox="0 0 512 512"
    width={25}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M105 162.2c-5.6 2.9-8.1 6.4-8.7 12.1-.4 3.5-.1 5.4 1.5 8.1 1.7 2.9 112.7 131.4 135.5 156.9 8.9 9.9 12.9 12.1 22.2 12.2 13.1 0 8.4 4.7 86.7-85.3 38.5-44.3 71-82 72.1-83.9 1.5-2.6 1.8-4.5 1.4-8-.6-5.7-3.1-9.2-8.7-12.1l-4.4-2.2H109.4l-4.4 2.2z"
      fillRule="nonzero"
    />
  </svg>
</button>
`;
